modules/wordpress-com-compose.php

```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+ * Module Name: WordPress.com Block Editor
+ * Module Description: Allow new block editor posts to be composed on WordPress.com.
+ * Jumpstart Description: Allow new block editor posts to be composed on WordPress.com.
+ * Sort Order: 15
+ * First Introduced: 7.2
+ * Requires Connection: Yes
+ * Auto Activate: No
+ * Module Tags: Writing
+ * Feature: Writing
+ * Additional Search Queries: iframes, allow, compose, WordPress.com, block, editor, post
+ */
+
+function jetpack_disable_send_frame_options_header() {
+	if ( jetpack_framing_allowed() ) {
+		remove_action( 'admin_init', 'send_frame_options_header' );
+	}
+}
+add_action( 'admin_init', 'jetpack_disable_send_frame_options_header', 1 ); // High priority to get ahead of send_frame_options_header
+
+function jetpack_framing_allowed() {
+	if ( ! empty( $_GET['frame-nonce'] ) && false !== strpos( $_GET['frame-nonce'], '.' ) ) {
+		list( $token, $signature ) = explode( '.', $_GET['frame-nonce'] );
+
+		$verified = Jetpack::init()->verify_xml_rpc_signature( $token, $signature );
```

Comment:
This will likely produce some warnings in https://github.com/Automattic/jetpack/blob/c8e0a2e3c6c5c045f027987a1facf4342ee6d7a4/class.jetpack.php#L5556.

Reply:
Fixed in 9aec65221.